docs/user-guide/expressions/casting.md

```diff
@@ -1,3 +1,104 @@
+
+
+The `cast` method includes a `strict` parameter that determines how Polars behaves when it encounters a value that can't be converted from the source `DataType` to the target `DataType`. By default, `strict=True`, which means that Polars will throw an error to notify the user of the failed conversion and provide details on the values that couldn't be cast. On the other hand, if `strict=False`, any values that can't be converted to the target `DataType` will be quietly converted to `null`.
+
+## Numerics
+
+Let's take a look at the following `DataFrame` which contains both integers and floating numbers.
+
+{{code_block('user-guide/expressions/casting','dfnum',['DataFrame'])}}
+
+```python exec="on" result="text" session="user-guide/cast"
+--8<-- "python/user-guide/expressions/casting.py:setup"
+--8<-- "python/user-guide/expressions/casting.py:dfnum"
+```
+
+To perform casting operations between floats and integers, or vice versa, we can invoke the `cast()` function.
+
+{{code_block('user-guide/expressions/casting','castnum',['cast'])}}
+
+```python exec="on" result="text" session="user-guide/cast"
+--8<-- "python/user-guide/expressions/casting.py:castnum"
+```
+
+Note that in the case of decimal values these are rounded downwards when casting to an integer.
+
+
+##### Downcast
+
+Reducing the memory footprint is also achievable by modifying the number of bits allocated to an element. As an illustration, the code below demonstrates how casting from `Int64` to `Int16` and from `Float64` to `Float32` can be used to lower memory usage.
+
+{{code_block('user-guide/expressions/casting','downcast',['cast'])}}
+
+```python exec="on" result="text" session="user-guide/cast"
+--8<-- "python/user-guide/expressions/casting.py:downcast"
+```
+
+#### Overflow
+
+When performing downcasting, it is crucial to ensure that the chosen number of bits (such as 64, 32, or 16) is sufficient to accommodate the largest and smallest numbers in the column. For example, using a 32-bit signed integer (`Int32`) allows handling integers within the range of -2147483648 to +2147483647, while using `Int8` covers integers between -128 to 127. Attempting to cast to a `DataType` that is too small will result in a `ComputeError` thrown by Polars, as the operation is not supported.
+
+{{code_block('user-guide/expressions/casting','overflow',['cast'])}}
+
+```python exec="on" result="text" session="user-guide/cast"
+--8<-- "python/user-guide/expressions/casting.py:overflow"
+```
+
+You can set the `strict` parameter to `False`, this converts values that are overflowing to null values.
+
+{{code_block('user-guide/expressions/casting','overflow2',['cast'])}}
+
+```python exec="on" result="text" session="user-guide/cast"
+--8<-- "python/user-guide/expressions/casting.py:overflow2"
+```
+
+## Strings
+
```

Comment:
Not per-se for this PR, but this might be a good page to introduce the string-cache categoricals cast as well?

Reply:
Yes, I was thinking about this. Right now AFAIK we don't really cover the categorical datatype at all.
Would prefer to add these at a later stage at the same time we have a proper introduction on the datatype.

Reply:
Yes, it could indeed have its own section as there certainly is some complexity involved. We could later backlink that section here I guess.